src/apify/apify_storage_client/_dataset_client.py

```diff
@@ -1,190 +1,183 @@
-        self._client = apify_dataset_client
 
-    @override
-    async def get(self) -> DatasetMetadata | None:
-        result = await self._client.get()
-        return DatasetMetadata.model_validate(result) if result else None
+class ApifyDatasetClient(DatasetClient):
+    """An Apify platform implementation of the dataset client."""
 
-    @override
-    async def update(
+    def __init__(
         self,
         *,
-        name: str | None = None,
-    ) -> DatasetMetadata:
-        return DatasetMetadata.model_validate(
-            await self._client.update(
-                name=name,
-            )
+        id: str,
+        name: str | None,
+        created_at: datetime,
+        accessed_at: datetime,
+        modified_at: datetime,
+        item_count: int,
+        api_client: DatasetClientAsync,
+    ) -> None:
+        """Initialize a new instance.
+
+        Preferably use the `ApifyDatasetClient.open` class method to create a new instance.
+        """
+        self._metadata = DatasetMetadata(
+            id=id,
+            name=name,
+            created_at=created_at,
+            accessed_at=accessed_at,
+            modified_at=modified_at,
+            item_count=item_count,
         )
 
+        self._api_client = api_client
+        """The Apify dataset client for API operations."""
+
+        self._lock = asyncio.Lock()
+        """A lock to ensure that only one operation is performed at a time."""
+
     @override
-    async def delete(self) -> None:
-        await self._client.delete()
+    @property
+    def metadata(self) -> DatasetMetadata:
+        return self._metadata
 
     @override
-    async def list_items(
-        self,
+    @classmethod
+    async def open(
+        cls,
         *,
-        offset: int | None = 0,
-        limit: int | None = BaseDatasetClient._LIST_ITEMS_LIMIT,  # noqa: SLF001
-        clean: bool = False,
-        desc: bool = False,
-        fields: list[str] | None = None,
-        omit: list[str] | None = None,
-        unwind: str | None = None,
-        skip_empty: bool = False,
-        skip_hidden: bool = False,
-        flatten: list[str] | None = None,
-        view: str | None = None,
-    ) -> DatasetItemsListPage:
-        return DatasetItemsListPage.model_validate(
-            vars(
-                await self._client.list_items(
-                    offset=offset,
-                    limit=limit,
-                    clean=clean,
-                    desc=desc,
-                    fields=fields,
-                    omit=omit,
-                    unwind=unwind,
-                    skip_empty=skip_empty,
-                    skip_hidden=skip_hidden,
-                    flatten=flatten,
-                    view=view,
-                )
-            )
+        id: str | None,
+        name: str | None,
+        configuration: Configuration,
+    ) -> ApifyDatasetClient:
+        token = configuration.token
+        api_url = configuration.api_base_url
+
```

Comment:
I believe this should use `configuration.default_dataset_id` if both `id` and `name` are `None`.